llama_index/llms/openrouter.py

```diff
@@ -0,0 +1,69 @@
+from typing import Any, Dict, Optional
+
+from llama_index.bridge.pydantic import Field
+from llama_index.constants import (
+    DEFAULT_CONTEXT_WINDOW,
+    DEFAULT_NUM_OUTPUTS,
+    DEFAULT_TEMPERATURE,
+)
+from llama_index.llms.generic_utils import get_from_param_or_env
+from llama_index.llms.openai import OpenAI
+from llama_index.llms.types import LLMMetadata
+
+DEFAULT_API_BASE = "https://openrouter.ai/api/v1"
+DEFAULT_MODEL = "gryphe/mythomax-l2-13b"
+
+
+class OpenRouter(OpenAI):
```

Comment:
should OpenRouter be a subclass of OpenAI?
if it is based on other LLMs, could we do something standalone like openai_like.py

Reply:
yea, just inherited openai like (needs the extra hook for api key)